great_expectations/core/batch_definition.py

```diff
@@ -69,14 +69,8 @@ def get_batch(self, batch_parameters: Optional[BatchParameters] = None) -> Batch
         Returns:
             A Batch of data.
         """
-        batch_list = self.data_asset.get_batch_list_from_batch_request(
-            self.build_batch_request(batch_parameters=batch_parameters)
-        )
-
-        if len(batch_list) == 0:
-            raise ValueError("No batch found")  # noqa: TRY003
-
-        return batch_list[-1]
```

Comment:
The asset is already doing the sorting, so moving the logic about picking the last batch to the asset as well.